### ISSUE TITLE
Bug when rendering the last page of a huge table (>950k records)

### ISSUE THREAD

**Issue by ydydxhg** [primefaces/primereact#755]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```
**Current behavior**
A dataTable with infinite scroll/virtual scroll implementation. When the table is very large, in my case > 950k records, and you scroll right to the bottom of the table. The last few rows appear to overlap. and left a gap between the last row to the bottom of the table, see below:
![image](https://user-images.githubusercontent.com/2774417/51880916-01b71b80-23de-11e9-8a28-41a006efbb4c.png).

However, the row selection still happens at the correct position, please see below, if i mouse over the blank space with the cross, the row highlighted gets the mouse style, and if i click at the crossed blank space, the correct row selection event gets triggered.
![image](https://user-images.githubusercontent.com/2774417/51881017-6b372a00-23de-11e9-815f-00b9ce2ae518.png)

so it gives me an impression that the html elements are stacked correctly but the rendering is screwed.

**Expected behavior**
the rows should appear correctly 

**Minimal reproduction of the problem with instructions**
need a large table that exceeds 950k records

**Please tell us about your environment:**
IIS express on local dev, azure app service iis in test environment. 

* **React version:**
6.4.1

* **PrimeReact version:**
2.0.1

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
chrome 71.0.3578.98 
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by mertsincan** [comment]
Could you please attach a codesandbox link for us to replicate?

**Comment by ydydxhg** [comment]
Hi please see my code:
`
const totalRecords = 10000;
const longTable = 'longTable';
class SearchForm extends Component {
  constructor(props) {
    super(props);
    let height = window.innerHeight - 240;
    if (this.props.hideSearch || this.props.subTab) {
      height -= 100;
    }
    // Firefox 1.0+
    let isFirefox = typeof InstallTrigger !== 'undefined';
    // Chrome 1+
    let isChrome = !!window.chrome && !!window.chrome.webstore;
    // Internet Explorer 6-11
    let isIE = /*@cc_on!@*/ false || !!document.documentMode;
    // Edge 20+
    let isEdge = !isIE && !!window.StyleMedia;

    this.state = {
      data: [],
      searchTerm: '',
      lazyTotalRecords: totalRecords,
      loading: false,
      rowPerPage: 60,
      emptyMessage: 'No records found',
      className: longTable,
      first: 0,
      height: height + 'px',
      selectedRow: null,
      searchClassNames: classNames({
        'p-inputgroup': !this.props.hideSearch,
        searchBar: !this.props.hideSearch,
        hidden: this.props.hideSearch,
      }),
      sortField: this.props.defaultOrder,
      sortOrder: this.props.defaultSort ? this.props.defaultSort : 1,
      rowHeight: this.props.rowHeight ? this.props.rowHeight : 35,
      isTableTooLong: false,
      isEdge: isEdge,
      isFirefox: isFirefox,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.clearSearch = this.clearSearch.bind(this);
    this.onVirtualScroll = this.onVirtualScroll.bind(this);
    this.onSort = this.onSort.bind(this);
    this.onRowClicked = this.onRowClicked.bind(this);
  }

  componentDidMount() {
    this.setState({
      searchUrl: this.props.searchUrl,
      loading: true,
    });

    let { defaultOrder, defaultSort } = this.props;
    if (defaultOrder && defaultSort) {
      this.search(null, 0, 20, defaultOrder, defaultSort);
    } else {
      this.search(null, 0, 20);
    }
  }

  componentDidUpdate() {
    if (this.state.searchUrl !== this.props.searchUrl) {
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState({
        searchUrl: this.props.searchUrl,
      });
      let { defaultOrder, defaultSort } = this.props;
      if (defaultOrder && defaultSort) {
        this.search(null, 0, 20, defaultOrder, defaultSort);
      } else {
        this.search(null, 0, 20);
      }
    }
  }

  onRowClicked(event) {
    const { allowClick, clickFunction } = this.props;
    if (allowClick && clickFunction) {
      clickFunction(event.value);
    }
  }
  onVirtualScroll(event) {
    this.setState({
      searchUrl: this.props.searchUrl,
      loading: true,
    });
    this.search(
      this.state.searchTerm,
      event.first,
      this.state.rowPerPage,
      this.state.sortField,
      this.state.sortOrder
    );
  }
  onSort(e) {
    this.setState(
      {
        sortField: e.sortField,
        sortOrder: e.sortOrder,
        loading: true,
      },
      () => {
        this.search(
          this.state.searchTerm,
          this.state.first,
          this.state.rowPerPage,
          e.sortField,
          e.sortOrder
        );
      }
    );
  }

  getTableHeightLimit() {
    /*
    Each browser can only handel certain height in a page,
    https://stackoverflow.com/questions/16637530/whats-the-maximum-pixel-value-of-css-width-and-height-properties

    Firefox: 33554400px
    Chrome:  33554428px
    Opera:   33554428px
    IE 9:    21474836.47px
    Edge: 10737418.23px
    */

    let tableHeightLimit = 33554428; //chrome is default
    if (this.state.isEdge) {
      tableHeightLimit = 10737417;
    } else if (this.state.isFirefox) {
      tableHeightLimit = 33554400;
    }
    return tableHeightLimit;
  }

  search(searchTerm = '', skip = 0, take = 45, order = null, sortOrder = 1) {
    var searchCondition = this.props.searchUrl + some_parameters; //normal rest API with paging parameters
    const client = new BaseApiClient(); //this is simply a custom built wrapper of axios with some header settings

    client.get(
      searchCondition,
      res => {
        if (!res || !res.data) {
          return;
        }
        const result = res;
        if (result.count === 0) {
          this.setState(
            {
              loading: false,
              first: 0,
              data: [],
              lazyTotalRecords: 1, //another display issue, if i dont set records count to 1, the empty message will not display when no record found...it will just display the header columns
              emptyMessage: 'No records found',
              className: '',
            },
            this.ensureTableHeight()
          );
          return;
        }

        var tableClass = '';
        this.setState(
          {
            loading: false,
            lazyTotalRecords: result.count + 1, // the + 1 here is another issue, if i only found 1 record, it wont display a full row's height, but rather show a tiny scroll bar that i have to click for one single row.
            data: result.data, 
            first: skip,
            className: tableClass,
          },
          this.ensureTableHeight()
        );
      },
      error => {
		  //error handling code removed here....
		
          this.ensureTableHeight();
        );
        console.log(JSON.stringify(error));
      },
      null, //this is just for other js logic, irrelavent to our display problem here, please ignore
      this.props.history  //this is just for routing, irrelavent to our display problem here, please ignore
    );
  }

  ensureTableHeight() {
    let isEdge = this.state.isEdge;
    let isFirefox = this.state.isFirefox;
    let tableHeightLimit = this.getTableHeightLimit();
    setTimeout(() => {
      let scrollBar = document.getElementsByClassName(
        'p-datatable-virtual-scroller'
      );
      let scrollHeight = scrollBar[0].style.height;

      let scroll = parseInt(scrollHeight.substring(0, scrollHeight.length - 2));

      const className = isFirefox
        ? 'superLongTableFF'
        : isEdge
        ? 'superLongTableEdge'
        : 'superLongTableChrome';
      if (scroll > tableHeightLimit) {
        scrollBar[0].classList.add(className);
        this.setState({ isTableTooLong: true });
      } else {
        scrollBar[0].classList.remove(className);
        this.setState({ isTableTooLong: false });
      }
    }, 1000); //we need to wait a bit for the css to finish calculating
  }

  handleChange(event) {
    this.setState({ searchTerm: event.target.value });
  }

  handleSubmit(event) {
    this.setState({
      loading: true,
    });
    this.search(
      this.state.searchTerm,
      0,
      this.state.rowPerPage,
      this.state.sortField,
      this.state.sortOrder
    );
    event.preventDefault();
  }

  clearSearch(event) {
    event.preventDefault();
    this.setState({ searchTerm: '', loading: true });
    let { defaultOrder, defaultSort } = this.props;
    if (defaultOrder && defaultSort) {
      this.search(null, 0, 20, defaultOrder, defaultSort);
    } else {
      this.search(null, 0, 20);
    }
  }
  render() {
    const dynamicColumns = this.props.columns.map((col, i) => {
        return (
          <Column
            key={col.field}
            field={col.field}
            header={col.header}
            sortable={col.sortable}
            className={col.columnClass}
          />
        );
    });
    return (
      <div
        className={
          this.props.hideSearch || this.props.subTab ? '' : styles.formContainer
        }>
        <form onSubmit={this.handleSubmit}>
          <Messages ref={el => (this.messages = el)} />
          <div className={this.state.searchClassNames}>
            <span className="p-inputgroup-addon searchAddOn">
              <i className="pi pi-search" />
            </span>
            <InputText
              value={this.state.searchTerm}
              onChange={this.handleChange}
              placeholder="Search"
              className="searchbox"
            />
            <Button label="Search" className="inputButton searchbutton" />
            <Button
              className="p-button-secondary inputButton"
              label="Clear"
              onClick={this.clearSearch}
            />
          </div>

          <div>
            <DataTable
              value={this.state.data}
              scrollable
              className={this.state.className}
              virtualScroll
              scrollHeight={this.state.height}
              rows={this.state.rowPerPage}
              totalRecords={this.state.lazyTotalRecords}
              lazy
              virtualRowHeight={this.state.rowHeight}
              selectionMode={this.props.allowClick ? 'single' : ''}
              selection={this.state.selectedRow}
              first={this.state.first}
              responsive
              onPage={this.onVirtualScroll}
              onVirtualScroll={this.onVirtualScroll}
              emptyMessage={this.state.emptyMessage}
              onSelectionChange={this.onRowClicked}
              loading={this.state.loading}
              sortField={this.state.sortField}
              sortOrder={this.state.sortOrder}
              onSort={this.onSort}>
              {dynamicColumns}
            </DataTable>
            {this.state.isTableTooLong ? (
              <span>
                We found a very large number of records, not all of the records
                are shown. Please search to narrow down the results
              </span>
            ) : (
              ''
            )}
          </div>
        </form>
      </div>
    );
  }
}
export default SearchForm;

`

**Comment by cagataycivici** [comment]
This is fixed in latest release, please try with 3.1.3
